src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,339 @@
+
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogon, "1");
+
+            SetValue(hive, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReason, "0");
+            SetValue(hive, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReasonUI, "0");
+
+            SetValue(hive, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeCaption, "");
+            SetValue(hive, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeText, "");
+        }
+
+        private void UpdateUserSpecificRegistrySettings(string domainName, string userName)
+        {
+            if (!_windowsServiceHelper.HasActiveSession(domainName, userName))
+            {
+                string securityId = _windowsServiceHelper.GetSecurityId(domainName, userName);
+                UpdateRegistrySettingsForDifferentUser(securityId);                
+            }
+            else
+            {
+                UpdateRegistrySettingsForCurrentUser();
+            }
+        }
+
+        private void UpdateRegistrySettingsForCurrentUser()
+        {
+            var hive = RegistryHive.CurrentUser;
+            SetValue(hive, RegistryConstants.SubKeys.ScreenSaver, RegistryConstants.ValueNames.ScreenSaver, "0");
+            SetValue(hive, RegistryConstants.SubKeys.ScreenSaverDomainPolicy, RegistryConstants.ValueNames.ScreenSaver, "0");
+            SetValue(hive, RegistryConstants.SubKeys.StartupProcess, RegistryConstants.ValueNames.StartupProcess, GetStartupCommand());
+        }
+
+        private void UpdateRegistrySettingsForDifferentUser(string securityId)
+        {
+            var hive = RegistryHive.Users;
+            SetValue(hive, $"{securityId}\\{RegistryConstants.SubKeys.ScreenSaver}", RegistryConstants.ValueNames.ScreenSaver, "0");
+            SetValue(hive, $"{securityId}\\{RegistryConstants.SubKeys.ScreenSaverDomainPolicy}", RegistryConstants.ValueNames.ScreenSaver, "0");
+            SetValue(hive, $"{securityId}\\{RegistryConstants.SubKeys.StartupProcess}", RegistryConstants.ValueNames.StartupProcess, GetStartupCommand());
+        }
+
+        private string GetStartupCommand()
+        {
+            //startup process            
+            string cmdExePath = System.Environment.GetEnvironmentVariable("comspec");
+            if (string.IsNullOrEmpty(cmdExePath))
+            {
+                cmdExePath = "cmd.exe";
+            }
+
+            //file to run in cmd.exe
+            var filePath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Root), "run.cmd");
+            
+            //extra "" are to handle the spaces in the file path (if any)
+            var startupCommand = $@"start ""Agent with AutoLogon"" {cmdExePath} /D /S /C """"{filePath}""""";
```

Comment:
Don't lead with `start`. Since it's not a rooted path, I believe Windows is actually going to actually launch `cmd.exe /c {cmdline you specified}` which is going to introduce confusion around quoting (and potentially quote stripping problems). Note, start is a shell-built-in (so there is no option to specify the full path to start).

Please change the command line to:
`$@"{cmdExePath} /D /S /C ""start ""Agent with AutoLogon"" ""{filePath}"" /startupType autologon"""`